### PR TITLE
[Data Objects] Relation Data-Types: Validation error if object without an ID was assigned

### LINKED ISSUE #5953: [Data Objects] Relation Data-Types: Validation error if object without an ID was assigned
The following example should throw an Exception: 
```php
$newObject = new MyDataObject(); 
$existingObject->setMyRelations([$newObject]);
$existingObject->save(); 
```
At the moment a `0` value is stored for `dest_id` in the relation table.

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves  #5953

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Relations/AllowAssetRelationTrait.php [hunk 1]
```diff
@@ -16,6 +16,10 @@ trait AllowAssetRelationTrait
      */
     protected function allowAssetRelation($asset)
     {
+        if ($asset instanceof Asset && $asset->getId() <= 0) {
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Relations/AllowAssetRelationTrait.php:19]: instanceof check really needed? as already done in checkValidity

Comment (weisswurstkanone) [reply models/DataObject/ClassDefinition/Data/Relations/AllowAssetRelationTrait.php:19]: I would keep it for consistency as the same is not true for objects. Besides that, this is not an internal method.

Comment (dvesh3) [reply models/DataObject/ClassDefinition/Data/Relations/AllowAssetRelationTrait.php:19]: then better would be `!$asset instanceof Asset || $asset->getId() <= 0`

Comment (weisswurstkanone) [reply models/DataObject/ClassDefinition/Data/Relations/AllowAssetRelationTrait.php]: done

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**